src/main/scala/com/datastax/gatling/plugin/model/DseCqlStatements.scala

```diff
@@ -98,40 +112,43 @@ case class DseCqlBoundStatementWithPassedParams(cqlTypes: CqlPreparedStatementUt
   */
 case class DseCqlBoundStatementWithParamList(cqlTypes: CqlPreparedStatementUtil,
                                              preparedStatement: PreparedStatement,
-                                             sessionKeys: Seq[String]) extends DseCqlStatement {
+                                             sessionKeys: Seq[String])
+  extends DseCqlStatement[BoundS, BoundB] {
 
   /**
     * Apply the Gatling session params to the Prepared statement
     *
     * @param gatlingSession DseSession
     * @return
     */
-  def buildFromSession(gatlingSession: Session): Validation[BoundStatement] = {
-    bindParams(gatlingSession, preparedStatement.bind(), sessionKeys).success
+  def buildFromSession(gatlingSession: Session): Validation[BoundB] = {
+    val template:BoundS = bindParams(gatlingSession, preparedStatement.bind(), sessionKeys)
+    new BoundB(template).success
   }
 
   /**
     * Bind the Gatling session params to the CQL Prepared Statement
     *
     * @param gatlingSession Gatling Session
-    * @param boundStatement CQL Prepared Statement
+    * @param template CQL Prepared Statement
     * @param sessionKeys    List of session params to apply, put in order of query ?'s
     * @return
     */
-  protected def bindParams(gatlingSession: Session, boundStatement: BoundStatement,
-                           sessionKeys: Seq[String]): BoundStatement = {
-
+  protected def bindParams(gatlingSession: Session, template: BoundS,
+                           sessionKeys: Seq[String]): BoundS = {
     val params = cqlTypes.getParamsList(preparedStatement)
-    var cnt = 0
-
-    sessionKeys.foreach { gatlingSessionKey =>
-      cqlTypes.bindParamByOrder(gatlingSession, boundStatement, params(cnt), gatlingSessionKey, cnt)
-      cnt += 1
+    val completedBuilder =
+      sessionKeys.foldLeft((0,new BoundB(template))) {
```

Comment:
maybe it will be more readable to use scala range from 0 and in the map step do parameters bind?

Reply:
You could do that, but then you wind up with a side-effecting map fn... which isn't awesome.

I agree with you that the existing usage is pretty ugly, though.  I think I can clean that up...